compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp

```diff
@@ -177,11 +177,64 @@ extern "C" void LLVMTimeTraceProfilerFinish(const char* FileName) {
 GEN_SUBTARGETS
 #undef SUBTARGET
 
+// Thanks to https://gist.github.com/dabrahams/1528856
+namespace access_retrofit {
```

Comment:
Is this change nice-to-have, or required? If it's just nice-to-have, I'd rather it be split into a separate PR, since I'm not personally comfortable reviewing this alone.

Reply:
I wrote more about this in https://github.com/rust-lang/rust/pull/111891#issuecomment-1562122922 and https://github.com/rust-lang/rust/pull/111891/commits/7161c5e7869acb664dfae55e375504d336e0de87 : prior to this change, it was impossible to introduce riscv UI tests to the compiler; I'm not super confident either in whatever is happening here, but it does seem important to me to include in this context. Could we ask someone else to help us take a look here?

Reply:
Let's tag @nikic and @rust-lang/wg-llvm 

It might still be worth to split this into a separate PR that can have a separate reviewer assigned.

Reply:
This is obviously not acceptable. You probably no longer need this because LLVM 14 is no longer supported.

Reply:
Ah! Dropping LLVM 14 support may indeed decrease the urgency here, though I'm not totally comfortable abandoning it entirely. The side-effect-ful filter call will cause any architecture's UI tests to fail any time any supported version of LLVM doesn't recognize a feature that rustc knows about, thereby seeming to defeat the purpose of the feature filter (at least as far as UI tests go)?

Reply:
Thanks for removing this change here. If you feel strongly about this, a separate PR is likely the right way to go about this.

Reply:
Well, the tests pass, so the good news is that means there's currently no features at the LLVM level that exist in any newer versions, which does indeed mean these changes can be decoupled. I did want to leave it up to @nikic what the resolution ought to be: I searched for a few hours before coming up with the approach here, and this was the only one I found that worked against system-provided LLVM. I'm open to an alternative approach, but if y'all would rather do nothing, then perhaps an issue documenting the problem would be a more fitting outcome than a PR with this change in it. What do you think?

Reply:
Opening an issue to document the problem is a reasonable approach to move forward with this.